src/schema/quiz/quiz.model.ts

```diff
@@ -0,0 +1,71 @@
+// Libraries
+import {Field, ObjectType, ID, GraphQLISODateTime} from 'type-graphql';
+import {
+  prop as Property,
+  getModelForClass,
+  modelOptions,
+} from '@typegoose/typegoose';
+import {ObjectId} from 'mongodb';
+
+// Models
+import {Question, QuestionModel} from '../question/question.model';
+
+@modelOptions({options: {allowMixed: 0}})
+@ObjectType({description: 'The Quiz model'})
+export class Quiz {
+  @Field(() => ID, {description: 'Quiz MongoDB ObjectID'})
+  id: ObjectId;
+
+  @Property({required: true, trim: true})
+  @Field({description: 'Name of the Quiz'})
+  name: string;
+
+  @Property({required: true})
+  @Field(() => GraphQLISODateTime, {description: 'Start time of the Quiz'})
+  startTime: Date;
+
+  @Property({required: true})
+  @Field(() => GraphQLISODateTime, {description: 'End time of the Quiz'})
+  endTime: Date;
+
+  @Property({default: []})
+  @Field(() => [String], {
+    description: 'An array containing the IDs of the questions',
+    name: 'questionIds',
+  })
+  questions: string[];
+
+  @Field(() => [Question], {
+    description:
+      'An array containing the details of all the questions in this quiz.',
+    name: 'questions',
+  })
+  async questionsArray(): Promise<(Question | null)[]> {
+    try {
+      return await Promise.all(
+        this.questions.map(questionId => QuestionModel.findById(questionId)),
+      );
+    } catch (error) {
+      return error;
+    }
+  }
+
+  @Property({default: []})
+  @Field(() => [String], {
+    description: 'A list of all the instructions for the quiz',
+  })
+  instructions: string[];
+
+  @Property({default: []})
+  @Field(() => [String], {
+    description:
+      'An array of objects containing the student id and corresponding marks',
+  })
+  submissions: {id: string; marks: number}[];
```

Comment:
We are storing students I'd for a single quiz twice, once here and other in user model for quiz id's array

Reply:
Yes agreed.
- In the user model, we are storing the list of quiz ids the user has attempted.
- Here we are saving the users' IDs who submitted the quiz, along with that we are also storing its corresponding marks. If we remove the IDs there is no relation between the marks and IDs stored in the DB.